lib/Sema/ConstraintSystem.cpp

```diff
@@ -4574,6 +4577,57 @@ SolutionResult ConstraintSystem::salvage() {
       }
     }
 
+    if (fixes.empty())
+      return SolutionResult::forUndiagnosedError();
+
+    {
+      auto &Diags = getASTContext().Diags;
+      DiagnosticTransaction fallbackErrors(Diags);
+
+      auto diagnoseSingleError = llvm::all_of(fixes,
+          [](auto &s) {
+            return s.Fixes.size() == 1;
```

Comment:
I think this should also account for warning fixes?

Reply:
Ah yes good catch!